vb-fe/src/components/roomModal/RoomModal.tsx

```diff
@@ -22,39 +20,58 @@ import {
   Text,
 } from "@chakra-ui/react";
 
-import "components/create/CreateRoom.scss";
 import { Select } from "components/select/Select";
 import { RoomForm } from "util/Room";
 import { Tag } from "util/Tags";
-import { useDispatch } from "react-redux";
-import { createRoomAction } from "_store/room/RoomActions";
+import { useDispatch, useSelector } from "react-redux";
+import { createRoomAction, updateRoomAction } from "_store/room/RoomActions";
 import { SystemConstants } from "_store/system/SystemTypes";
+import { State } from "_store/rootReducer";
 
 const MAX_NAME_LENGTH = 40;
 
 type Props = {
   open: boolean;
   handleError: (error: any) => void;
+  currentRoom?: {
+    _id: string;
+    name: string;
+    description: string;
+    tags: Tag[];
+  };
 };
 
 interface RoomType extends RoomForm {
   error: boolean;
 }
 
-const initialRoomValues: RoomType = {
-  name: "",
-  description: "",
-  tags: [],
-  error: false,
-};
+export const RoomModal: FunctionComponent<Props> = ({ open, handleError, currentRoom }) => {
+  const initialRoomValues: RoomType = {
+    name: currentRoom?.name || "",
+    description: currentRoom?.description || "",
+    tags: currentRoom?.tags || [],
+    error: false,
+  };
 
-export const CreateRoom: FunctionComponent<Props> = ({ open, handleError }) => {
   const dispatch = useDispatch();
+  const modalType = useSelector((state: State) => state.system.roomModal.type);
   const [room, setRoom] = useState(initialRoomValues);
 
-  /**
-   * Validates the form to see if the room name is present
-   */
+  let buttonText;
+  let headerText;
```

Comment:
change `headerText` and `buttonText` to useState, and move the switch to a `useEffect`.

For the default case throw an error - `throw new Error('Invalid room type')`